tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -87,7 +87,16 @@ func (rb *IptablesBuilderImpl) AppendRuleV6(chain string, table string, params .
 
 func (rb *IptablesBuilderImpl) BuildV4() []string {
 	rules := []string{}
+	chainTableLookupMap := make(map[string]bool)
 	for _, r := range rb.rules.rulesv4 {
+		chainTable := fmt.Sprintf("%s:%s", r.chain, r.table)
+		// Create new chain if key: `chainTable` isn't present in map
+		if _, present := chainTableLookupMap[chainTable]; !present {
+			cmd := fmt.Sprint(strings.Join(
```

Comment:
You have to exclude the standard chains (`INPUT`, `OUTPUT`, etc.).
I'd define a map in a global variable with those strings, so it's easy to lookup.